tests/fixtures/ead/ead_record_attribute_and_subfield_variations.xml

```diff
@@ -38,15 +38,21 @@
                         <unitdate datechar="creation"></unitdate>
                         <unitdate datechar="creation">1905-2012</unitdate>
                         <unitdate datechar="creation">1905</unitdate>
+                        <unitdate datechar="creation">abcd-efgh</unitdate>
```

Comment:
The test suite for this app is very confusing. I'm not asking for a specific change as part of this code review as I have confirmed that reverting the code in `ead.py` to the version without the conditional check fails the suite as expected. I'm sharing this input mostly as a bit of a warning that the process of testing may not be intuitive enough for others to assist in maintaining this application. There was no additional regression tests added to the suite to ensure the block of code works as intended, yet somehow adding additional lines to a single fixture file is allowing us ensure the code logic is correct.

For unit testing, I'd expect we would have added an additional test for the new regression such as:
`test dates should only be valid if there is a value or a range`

Making a test suite that is explicit like that allows us to read through the tests to understand functionality. Again, I'm not stating this test suite doesn't test the code as I believe it does. I'm just nervous it is a bit too opaque as to how the code is supposed to work even if the tests are passing "correctly".

Reply:
When I revert the code in ead.py, I get an error in this test (as expected):
`test_ead_record_with_attribute_and_subfield_variations_transforms_correctly`

It isn't clear _why_ the transform isn't working and I believe it will be very much harder to support this application with only "top level" testing rather than more granular tests... particularly when guarding against regressions.

Reply:
We discussed this before and I do agree. It would be a significant overhaul of the testing suite but I agree that it's worthwhile for the long-term maintainability. I will create a ticket to do that

Reply:
I agree that a full overhaul would take some work, but there isn't anything really preventing us from adding one or two additional tests when we introducing changes like these. The giant mega test AND small focused tests is genuinely the best of both worlds.

Reply:
Good call, I'll add the tests in a new commit